builtin/providers/azure/config.go

```diff
@@ -0,0 +1,63 @@
+package azure
+
+import (
+	"fmt"
+	"log"
+	"sync"
+
+	"github.com/Azure/azure-sdk-for-go/management"
+)
+
+// Config is a struct which holds all the required information to access
+// Azure services.
+type Config struct {
+	PublishSettingsFilePath string
+	SubscriptionID          string
+	ManagementCert          []byte
+	ManagementUrl           string
+}
+
+// AzureClient contains all the handles required for managing Azure services.
+type AzureClient struct {
+	// the client which holds the authentification information
+	managementClient management.Client
+
+	// unfortunately; because of how Azure's network API works; doing networking operations
+	// concurrently is very hazardous, and we need a mutex.
+	mutex *sync.Mutex
+}
+
+// Client configures and returns a fully initialized Azure client.
+func (c *Config) Client() (interface{}, error) {
+	var err error
+	var managementClient management.Client
+	var azureClient AzureClient
+
+	log.Println("[DEBUG] Building Azure management client.")
+	if c.PublishSettingsFilePath != "" {
+		managementClient, err = management.ClientFromPublishSettingsFile(
+			c.PublishSettingsFilePath,
+			c.SubscriptionID,
+		)
+	} else if c.ManagementUrl != "" {
```

Comment:
Using multiple `else if's` is not really idiomatic Go. So taking the comment above into consideration (about loosing the custom `AzureClient` type) it would be cleaner to just have something like (pseudo code):

```go
if c.PublishSettingsFilePath != "" {
  return management.ClientFromPublishSettingsFile(...)
}

if c.ManagementUrl != "" {
  return management.NewClientFromConfig(...)
}
```

And the additional var declaration at the beginning of this function could also be removed...